data-raw/ictr/ictr_main.Rmd

```diff
@@ -0,0 +1,293 @@
+    - sectors (as the matching with ecoinvent is not done yet, we do not have one sector per product yet)
+
+```{r}
+inputs <- read.csv(here("data_prep","02_input_data", "data_ecoinvent_co2.csv"))
+companies <- read.csv(here("data_prep","02_input_data", "tilt_companies.csv"))
+```
+
+# Step 2: Calculate ranks per input product and assign scores
+
+## 2.1 Calculate the rank of each input product
+
+This is done based on the input products' carbon footprints compared to 4
+different benchmarks (all products, products with same unit, products with same
+sector, products with same unit and sector)
+
+```{r}
+# rank in comparison to all input products
+ecoinvent_input <- inputs %>%
+  mutate(perc_all = rank(input_co2) / length(input_co2))
+
+# rank in comparison to all input products with same unit
+ecoinvent_input <- ecoinvent_input |>
+  group_by(unit) |>
+  mutate(perc_unit = rank(input_co2) / length(input_co2)) %>%
+  ungroup()
+
+# rank in comparison to all input products with same input sector
+ecoinvent_input <- ecoinvent_input |>
+  group_by(input_sector) |>
+  mutate(perc_sec = rank(input_co2) / length(input_co2)) %>%
+  ungroup()
+
+# rank in comparison to all input products with same unit and input sector
+ecoinvent_input <- ecoinvent_input |>
+  group_by(unit, input_sector) |>
+  mutate(perc_unit_sec = rank(input_co2) / length(input_co2)) %>%
+  ungroup()
+```
+
+## 2.2 Assign scores to the input products
+
+This is done based on their position within the distribution of co2 footprints
+in comparison to different benchmarks.
+
+```{r}
+# creat variables for thresholds
+low_threshold <- 0.3
+high_threshold <- 0.7
+
+# assign scores to position within percentile distribution
+# for all input products
+ecoinvent_scores <- ecoinvent_input %>%
+  mutate(
+    score_all = case_when(
+      perc_all < low_threshold ~ "low",
+      perc_all >= low_threshold & perc_all < high_threshold ~ "medium",
+      perc_all >= high_threshold ~ "high"
+    )
+  )
+
+# for products with same unit
+ecoinvent_scores <- ecoinvent_scores %>%
+  mutate(
+    score_unit = case_when(
+      perc_unit < low_threshold ~ "low",
+      perc_unit >= low_threshold & perc_unit < high_threshold ~ "medium",
+      perc_unit >= low_threshold ~ "high"
+    )
+  )
+
+# for products with same sector
+ecoinvent_scores <- ecoinvent_scores %>%
+  mutate(
+    score_sector = case_when(
+      perc_sec < low_threshold ~ "low",
+      perc_sec >= low_threshold & perc_sec < high_threshold ~ "medium",
+      perc_sec >= low_threshold ~ "high"
+    )
+  )
+
+# for products with same unit and sector
+ecoinvent_scores <- ecoinvent_scores %>%
+  mutate(
+    score_unit_sec = case_when(
+      perc_unit_sec < low_threshold ~ "low",
+      perc_unit_sec >= low_threshold & perc_unit_sec < high_threshold ~ "medium",
+      perc_unit_sec >= high_threshold ~ "high",
+    )
+  )
+
+rm(ecoinvent_input)
+rm(high_threshold)
+rm(low_threshold)
```

Comment:
Please remove the calls to `rm()`. 

> The problem with rm(list = ls()) is that, given the intent, it does not go far enough. All it does is delete user-created objects from the global workspace. 
-- https://rstats.wtf/save-source.html#rm-list-ls

So in general, `rm()` isn't useful and in particularly here I do need the every meaningful intermediate result to stay in the rendering-environment of the .Rmd file -- because I take a snapshots of then before I start refactoring the code. It's a fist coarse test that allows me to restructure the code a bit then write more specific tests.

Reply:
@maurolepore I totally get your point and will take care of it in the future